src/inmanta/deploy/scheduler.py

```diff
@@ -360,19 +361,31 @@ async def _build_resource_mappings_from_db(
 
         :return: resource_mapping {id -> resource details}
         """
```

Comment:
I feel like the changes to this method are mixing up responsibilities. I would say the scheduler is the authority on the resource state, and it ensures that the db comes to reflect that state. The db is the authority on the resource intent, which is what we should be fetching here.

I think we should return only resource id, attribute hash, attributes and either `const.ResourceStatus` or simply an `undefined` boolean. Whether we use the `ResourceDetails` dataclass, or a typed dict, or .... for it is an implementation detail. Either way, I do not think we should read anything other than resource intent from the db, nor return it, nor have a field for it in `ResourceDetails`.

Reply:
Exception: when we initialize the scheduler we could read the state as well, and use it to restore scheduler state (requires careful consideration of the scenario where the scheduler is stopped before it had a chance to process the latest version). But for now I think that the "increment calculation" we have going on will suffice.